core/Network.js

```diff
@@ -86,12 +87,20 @@ Network.prototype.getProcessByName = function (processName) {
 
 Network.prototype.run = function (runtime, options, callback) {
   options = options || {};
-
-  _.forEach(this._processes, function (process) {
-    _.invokeMap(process.inports, 'setRuntime', runtime);
-    _.invokeMap(process.outports, 'setRuntime', runtime);
-  });
-  runtime.run(_.values(this._processes), options, callback || function () {});
+  callback = callback || function () {};
+
+  _.invokeMap(this._connections, 'setRuntime', runtime);
+
+  try {
+    runtime.run(_.values(this._processes), options, callback );
+  } catch (e) {
+    console.log('Connections');
```

Comment:
Then runtime still displays the deadlock details. Network then catches the exception that gets thrown and shows the connection details.